// src/routes/automation.ts
// routes/automation.ts
import { RouteConfig } from '../routes/routes';

export const automationRoutes: RouteConfig[] = [
  {
    path: '/business-automation/sales-process-automation',
    file: 'business-automation/sales-process-automation.html',
    title: 'Sales Automation | CRM & Lead Management Philippines',
    description: 'Boost sales with CRM automation, lead tracking, and follow-ups. Drive higher conversions and streamline your pipeline.',
    keywords: ['sales automation', 'CRM software', 'lead management solutions', 'sales pipeline automation', 'sales process optimization', 'Philippines', 'USA', 'UK', 'Australia'],
    changeFreq: 'monthly',
    priority: '0.8',
    structuredData: {
      "@context": "https://schema.org",
      "@type": "Service",
      "serviceType": "Sales Process Automation",
      "provider": {
        "@type": "Organization",
        "name": "Tepa Solutions"
      },
      "description": "Sales automation solutions including CRM, lead management, and pipeline optimization to increase conversions.",
      "areaServed": [
        { "@type": "Country", "name": "United States" },
        { "@type": "Country", "name": "Philippines" },
        { "@type": "Country", "name": "United Kingdom" },
        { "@type": "Country", "name": "Australia" },
         { "@type": "Place", "name": "Worldwide" }
      ]
    }
  },
  {
    path: '/business-automation/marketing-automation',
    file: 'business-automation/marketing-automation.html',
    title: 'Marketing Automation | Email & Campaign Tools Philippines',
    description: 'Automate campaigns, emails, and lead nurturing. Improve engagement and ROI with smart marketing automation.',
    keywords: ['marketing automation', 'email automation', 'campaign automation tools', 'lead nurturing software', 'digital marketing automation', 'Philippines', 'USA', 'UK', 'Australia'],
    changeFreq: 'monthly',
    priority: '0.8',
    structuredData: {
      "@context": "https://schema.org",
      "@type": "Service",
      "serviceType": "Marketing Automation",
      "provider": {
        "@type": "Organization",
        "name": "Tepa Solutions"
      },
      "description": "Comprehensive marketing automation solutions for email campaigns, lead nurturing, and customer engagement.",
      "areaServed": [
        { "@type": "Country", "name": "United States" },
        { "@type": "Country", "name": "Philippines" },
        { "@type": "Country", "name": "United Kingdom" },
        { "@type": "Country", "name": "Australia" },
         { "@type": "Place", "name": "Worldwide" }
      ]
    }
  },
  {
    path: '/business-automation/customer-support-automation',
    file: 'business-automation/customer-support-automation.html',
    title: 'Customer Support Automation | AI Chatbots & Help Desk',
    description: 'Deliver 24/7 customer service with AI chatbots, ticketing, and automated workflows. Improve support efficiency.',
    keywords: ['customer support automation', 'AI chatbots', 'help desk software', 'support ticket automation', 'customer service solutions', 'Philippines', 'USA', 'UK', 'Australia'],
    changeFreq: 'monthly',
    priority: '0.8',
    structuredData: {
      "@context": "https://schema.org",
      "@type": "Service",
      "serviceType": "Customer Support Automation",
      "provider": {
        "@type": "Organization",
        "name": "Tepa Solutions"
      },
      "description": "AI-powered customer support automation with chatbots, help desk systems, and automated support workflows.",
      "areaServed": [
        { "@type": "Country", "name": "United States" },
        { "@type": "Country", "name": "Philippines" },
        { "@type": "Country", "name": "United Kingdom" },
        { "@type": "Country", "name": "Australia" },
         { "@type": "Place", "name": "Worldwide" }
      ]
    }
  },
  {
    path: '/business-automation/hr-automation',
    file: 'business-automation/hr-automation.html',
    title: 'HR Automation | Payroll & Employee Management Software',
    description: 'Automate payroll, recruitment, and HR workflows. Save time and manage employees efficiently.',
    keywords: ['HR automation', 'payroll software', 'employee management system', 'recruitment automation', 'HR process automation', 'Philippines', 'USA', 'UK', 'Australia'],
    changeFreq: 'monthly',
    priority: '0.8',
    structuredData: {
      "@context": "https://schema.org",
      "@type": "Service",
      "serviceType": "HR Process Automation",
      "provider": {
        "@type": "Organization",
        "name": "Tepa Solutions"
      },
      "description": "HR automation solutions covering recruitment, onboarding, payroll, and employee performance management.",
      "areaServed": [
        { "@type": "Country", "name": "United States" },
        { "@type": "Country", "name": "Philippines" },
        { "@type": "Country", "name": "United Kingdom" },
        { "@type": "Country", "name": "Australia" },
         { "@type": "Place", "name": "Worldwide" }
      ]
    }
  },
  {
    path: '/business-automation/finance-automation',
    file: 'business-automation/finance-automation.html',
    title: 'Finance Automation | Accounting & Invoicing Solutions',
    description: 'Automate accounting, invoicing, and financial reporting. Improve accuracy and reduce manual work.',
    keywords: ['finance automation', 'accounting automation software', 'invoice automation', 'financial reporting tools', 'automated accounting solutions', 'Philippines', 'USA', 'UK', 'Australia'],
    changeFreq: 'monthly',
    priority: '0.8',
    structuredData: {
      "@context": "https://schema.org",
      "@type": "Service",
      "serviceType": "Finance Process Automation",
      "provider": {
        "@type": "Organization",
        "name": "Tepa Solutions"
      },
      "description": "Financial process automation covering invoicing, expense tracking, compliance, and reporting to improve accuracy.",
      "areaServed": [
        { "@type": "Country", "name": "United States" },
        { "@type": "Country", "name": "Philippines" },
        { "@type": "Country", "name": "United Kingdom" },
        { "@type": "Country", "name": "Australia" },
         { "@type": "Place", "name": "Worldwide" }
      ]
    }
  },
  {
    path: '/business-automation/inventory-management-automation',
    file: 'business-automation/inventory-management-automation.html',
    title: 'Inventory Automation | Stock & Supply Chain Management',
    description: 'Automate inventory tracking, reorders, and supply chain. Reduce costs and prevent stockouts.',
    keywords: ['inventory automation', 'stock management software', 'inventory tracking system', 'supply chain automation', 'warehouse management solutions', 'Philippines', 'USA', 'UK', 'Australia'],
    changeFreq: 'monthly',
    priority: '0.8',
    structuredData: {
      "@context": "https://schema.org",
      "@type": "Service",
      "serviceType": "Inventory Management Automation",
      "provider": {
        "@type": "Organization",
        "name": "Tepa Solutions"
      },
      "description": "Automated inventory management solutions including stock tracking, reorder alerts, and supply chain optimization.",
      "areaServed": [
        { "@type": "Country", "name": "United States" },
        { "@type": "Country", "name": "Philippines" },
        { "@type": "Country", "name": "United Kingdom" },
        { "@type": "Country", "name": "Australia" },
         { "@type": "Place", "name": "Worldwide" }
      ]
    }
  }
];
